network/discovery/src/validator_set.rs

```diff
@@ -65,12 +65,15 @@ impl ValidatorSetStream {
             .set(mismatch);
     }
 
-    fn extract_updates(&mut self, payload: OnChainConfigPayload) -> PeerSet {
+    fn extract_updates(
+        &mut self,
+        payload: OnChainConfigPayload,
+    ) -> Result<PeerSet, DiscoveryError> {
         let _process_timer = EVENT_PROCESSING_LOOP_BUSY_DURATION_S.start_timer();
 
-        let node_set: ValidatorSet = payload
-            .get()
-            .expect("failed to get ValidatorSet from payload");
+        let node_set: ValidatorSet = payload.get().map_err(|err| {
+            DiscoveryError::Parsing(format!("Failed to get ValidatorSet from payload {}", err))
+        })?;
```

Comment:
How is this recoverable? My understanding is that this won't fail due to invalid network addresses.

Reply:
There are other things that you can stick in the validator set that won't process.

I think we need to think about how we handle if the `ConsensusPublicKey` is invalid.  I don't believe there's anything in move that ensures that it can be deserialized

Reply:
yeah it's a TODO to add checks for consensus public keys

Reply:
https://github.com/aptos-labs/aptos-core/pull/1604